MIP/mip-1/README.md

```diff
@@ -0,0 +1,113 @@
+- **Authors**: [Liam Monninger](mailto:liam@movementlabs.xyz)
+- **Desiderata**: [MD-1](../MD/md-1)
+
+## Abstract
+
+ENTL proposes a simple challenge-reponse protocol to link Enclave usage to a given program. It additionally proposes a means of time-locking to allow the replacement of said program while maintaining security. This approach is demonstrated to be both generalizable and cryptographically sound.
+
+## Motivation
+
+This protocol was originally suggested as a solution to minimizing trust assumptions in the Atomic Bridge under [MD-1](../MD/md-1). However, similar problems were encountered under Movement Labs Governance Stage 0 concerning general key usage. 
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Protocol Description
+The protocol is a simple challenge-response scheme performed between two parties: the Enclave, a presumed secure compute unit, and the Client, the program requesting the Enclave's services. 
+
+The following messages types are exchanged between the Enclave and the Client:
+- **SYN**: the Client sends a 256-bit nonce to the Enclave.
+- **SYN-OK**: the Enclave responds indicating it has received and accepted the nonce. This can be sent as a response to either SYN or SYN-CHECK.
+- **SYN-TL**: the Enclave responds indicating it has the received the nonce and subjected it to a time-lock and queuer. The time-lock and place in the queue are provided as response. 
+- **APP**: the Client sends an application message to the Enclave, a 256-bit nonce for the current message, and a 256-bit nonce for the next message.
+- **APP-OK**: the Enclave responds indicating it has processed the application message and has updated the nonce for the next message.
+- **APP-OK-CON**: the Enclave responds indicating that it has processed the application message, has updated the nonce for the next message, and has aborted a contentious synchronization attempt. 
+- **APP-REJ**: the Enclave responds indicating it has received the application message but is rejecting it and will not update the nonce for the next message.
+
+These messages are exchanged as follows:
+
+**Synchronization Phase**
+1. Prior to sending its first application message, the Client generates and safely stores CSPRNG nonce and sends it to the Enclave in a **SYN** message.
+2. The Enclave processes the **SYN** message:
+    - If enclave does not have a stored nonce, this indicates that no other client has attempted to synchronize with the Enclave. The Enclave MUST accept the nonce, store the nonce, and respond with a **SYN-OK** message.
+    - If the Enclave has a record of the nonce, this indicates that another client has attempted to synchronize with the Enclave in the past. It MUST check queue and time-lock conditions:
```

Comment:
It's not explained what clock should the enclave use to enforce the time lock.
The section on Time-Locking in practical considerations further states that enclaves typically rely on host clocks which may weaken security of the system.

Reply:
This is actually discussed here: https://github.com/movementlabsxyz/MIP/tree/l-monninger/entl/MIP/mip-1#time-locking

Reply:
I see that, but I can't see what exact approach that section recommends. On re-reading (but without getting into the weeds of the referenced paper) I understand that use of the host-provided clock as the sole source of truth is rejected in favor of performing computing work in the enclave with some reasonably predictable execution time, and the "hybrid approach" involves the enclave's host clock as a sort of BogoMIPS-style factor estimator. Is this correct?